From b4f89ed1deadd32b000d2cfb11763608acbccc93 Mon Sep 17 00:00:00 2001
From: Marius-Cristian Baciu <2309bmcristi@gmail.com>
Date: Sun, 19 Jul 2020 02:53:54 +0300
Subject: [PATCH] Solve stdio.d unittest bug

open_wmemstream() guarantees a NULL byte at the end of the buffer,
but the memcmp() call expects last 4 bytes to be NULL

Signed-off-by: Marius-Cristian Baciu <2309bmcristi@gmail.com>
---
 libphobos/libdruntime/core/sys/posix/stdio.d | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libphobos/libdruntime/core/sys/posix/stdio.d b/libphobos/libdruntime/core/sys/posix/stdio.d
index 0c6a144f5..7a070aaf6 100644
--- a/libphobos/libdruntime/core/sys/posix/stdio.d
+++ b/libphobos/libdruntime/core/sys/posix/stdio.d
@@ -485,7 +485,7 @@ unittest
     assert(f !is null);
     assert(fwprintf(f, testdata.ptr) == 5);
     assert(fflush(f) == 0);
-    assert(memcmp(ptr, testdata.ptr, testdata.length*wchar_t.sizeof) == 0);
+    assert(memcmp(ptr, testdata.ptr, (testdata.length - 1)*wchar_t.sizeof) == 0);
     assert(fclose(f) == 0);
 }
 
-- 
2.17.1

